### PR TITLE
Make mouse following animation easier to read.

### PR DESCRIPTION
**Pull request by Khongchai**
#### Summary

Refactored mouse following animation so that it's easier to read without changing the functionality.

#### Motivation
This example, as of now, is full of single-character variable names and a few unused functions. I understand that this code was, according to the original github repo, written as a POF and, understandably, not optimized. However, for learning purposes, making it easier to read in general would be more beneficial for everyone, especially beginners or those who are new to canvas. 

#### Supporting details
What I did:
1. Rewrote variable and function names like c, cc, a, gc, cn, etc. to be more self-explanatory.
2. Removed unused lines of code.
3. Moved script tag from top to bottom. 
4. window.onresize was removed; it was just generating random positions on every resize and causes flashing. 
5. window.onload removed and functions were moved outside.

All functionalities are the same after refactoring: https://codesandbox.io/s/mdn-mouse-following-edit-l5qd6?file=/index.html

#### Metadata

- [ ] Adds a new document
- [x ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Review by wbamberg — COMMENTED** [review]
Thanks for this PR! I haven't looked at it in a lot of detail but it seems to be a great improvement.

I think we should remove the external link to https://kunalverma94.github.io/gallery/gags/beyblade.html and make this into a "live sample" so we can show the output right in the page. To do this, you'd need to:

* Change the title of this section from "Mouse Following Animation" to "Mouse following animation" (this isn't at all needed, but we use sentence case for titles, and since we are going to reference the title here it's definitely worth fixing).

* separate out the HTML, CSS, and JavaScript into separate code blocks, using language identifiers on the Markdown code fences, like:

    \```js
    // the js goes here
    \```

* Before each code block, add an H3 heading identifying it, like:

    \### JavaScript

* After all the code blocks, add another H3 like:

    \### Result

* Right under that H3, add a line like:

    {{EmbedLiveSample("Mouse_following_animation", "500", "500")}}

    (you might need to adjust the second "500" argument to fit the height of the demo)

* Remove the "Output" H5 and the link underneath it.

After that I might have some more specific comments on the code, but it will be much easier to do that once we have the basic structure in place.

File: files/en-us/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 1]
```diff
@@ -320,364 +327,492 @@ Below is the {{HTMLElement("canvas")}} in which the image is scrolled. Note that
...
-        <canvas id="cw"></canvas>
-    </body>
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Document</title>
+    <style>
+      /* css here */
+    </style>
```
Comment (wbamberg) [inline files/en-us/web/api/canvas_api/tutorial/basic_animations/index.md:344]: You don't need this.

File: files/en-us/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 2]
```diff
@@ -320,364 +327,492 @@ Below is the {{HTMLElement("canvas")}} in which the image is scrolled. Note that
...
+  </head>
+  <body>
+    <canvas id="cw"></canvas>
+  </body>
+  <script>
+    //...javascript here
+  </script>
```
Comment (wbamberg) [inline files/en-us/web/api/canvas_api/tutorial/basic_animations/index.md:351]: Or this.

File: files/en-us/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 3]
```diff
@@ -320,364 +327,492 @@ Below is the {{HTMLElement("canvas")}} in which the image is scrolled. Note that
...
+
+body {
+  margin: 0;
+  padding: 0;
+  background-color: rgba(0, 0, 0, 0.05);
+}
+```
 
-<https://kunalverma94.github.io/gallery/gags/beyblade.html>
+### Javascript
```
Comment (wbamberg) [inline files/en-us/web/api/canvas_api/tutorial/basic_animations/index.md:370]:
```suggestion
### JavaScript
```

**Review by wbamberg — CHANGES_REQUESTED** [review]
This looks really great, a big improvement! I had a few small comments.

Also though, it looks like your editor is set up to run Prettier over files, so there are a lot of extra changes here. Do you think you could remove those so we just get the changes to this example? (I'm in favour of running Prettier over our examples (https://github.com/mdn/content/pull/9565), but think this should be its own project.)

> I would like to point out that this doesn't work very well on mobile (there's no touch listener, user cannot drag, only tap). Should something be done about that, or do you think it's already too much for just one example? I'd be happy to lend a hand and contribute a few more lines.

I think it's worth doing if it doesn't add a lot of size or complexity. If so I'd be happy to review a PR for this, if you want to have a go, but I think it should be in a follow-up.

Thanks again for your work on this example!

**Comment by wbamberg** [comment]
FWIW I also think we should remove the snake game from this page. It's too long and complex to be a good example, even if it were updated for readability. But also this should be a separate PR.

**Comment by Khongchai** [comment]
Ok, I just reverted the unintended prettier format on other sections and applied your requested changes. 

> FWIW I also think we should remove the snake game from this page. It's too long and complex to be a good example, even if it were updated for readability. But also this should be a separate PR.

Yes, I agree. I was shocked myself seeing the humongous code while I was going through the tutorial.

As for the mobile support, it will just be five more lines. But yes, let's do that in another PR.

**Review by wbamberg — APPROVED** [review]
Thanks for the update @Khongchai ! I realised I forgot to say: you only need to include the content of the `<body>` in a live sample. Rather than make you go round review again I just pushed a change for that, I hope you don't mind.

Thanks again for your work on this PR, it looks great.

**Comment by wbamberg** [comment]
> Ok, I just reverted the unintended prettier format on other sections and applied your requested changes.
> 
> > FWIW I also think we should remove the snake game from this page. It's too long and complex to be a good example, even if it were updated for readability. But also this should be a separate PR.
> 
> Yes, I agree. I was shocked myself seeing the humongous code while I was going through the tutorial.
> 
> As for the mobile support, it will just be five more lines. But yes, let's do that in another PR.

Thanks for your work on this. I'd be happy to review PRs for either of these additional things. Especially let me know if you *don't* want to remove the snake game and I'll take care of that.

**Comment by Khongchai** [comment]
Yes, you can take care of the snake game. I'll take a look at the touch event handler.

**Pull request merged**